src/openms/source/METADATA/ID/IdentificationData.cpp

```diff
@@ -93,23 +93,41 @@ namespace OpenMS
 
 
   IdentificationData::InputFileRef
-  IdentificationData::registerInputFile(const String& file)
+  IdentificationData::registerInputFile(const InputFile& file)
   {
-    return input_files_.insert(file).first;
+    if (!no_checks_ && file.name.empty()) // key may not be empty
+    {
+      String msg = "input file must have a name";
+      throw Exception::IllegalArgument(__FILE__, __LINE__,
+                                       OPENMS_PRETTY_FUNCTION, msg);
+    }
+    auto result = input_files_.insert(file);
+    if (!result.second) // existing element - merge in new information
```

Comment:
can you add a comment here what data get's merged?

Reply:
The idea (in IdentificationData in general) is that the merging gets delegated to the respective data classes. So IdentificationData itself doesn't know, and shouldn't need to know, what gets updated.

Reply:
Ok but maybe we need a central point to divinest this for all data. Maybe at the level of the ID datastructure because users might look there first.

Reply:
I've added a sentence to the IdentificationData documentation.